openssl-dynamic/src/main/java/org/apache/tomcat/jni/SSL.java

```diff
@@ -881,4 +883,56 @@ public static native void setCertificateBio(
      * Free x509 chain ({@code STACK_OF(X509)} pointer).
      */
     public static native void freeX509Chain(long x509Chain);
+    
+    /**
+     * This is the default (initial) status type value for {@link SSLEngine}s. It can be used
+     * to disable OCSP stapling on a {@link SSLEngine} once it has been enabled (but before
+     * the handshake has happened).
+     * 
+     * Search for it in the {@code tls1.h} file.
+     * 
+     * @see #TLSEXT_STATUSTYPE_ocsp
+     */
+    public static final int TLSEXT_STATUSTYPE_host_name = 0;
+    
+    /**
+     * Enables OCSP stapling on a {@link SSLEngine}.
+     * 
+     * https://www.openssl.org/docs/man1.0.2/ssl/SSL_set_tlsext_status_type.html
+     * 
+     * @see #setStatusType(long, int)
+     */
+    public static final int TLSEXT_STATUSTYPE_ocsp = 1;
+    
+    /**
+     * Enables or disables OCSP stapling on the given {@link SSLEngine}.
+     *
+     * https://www.openssl.org/docs/man1.0.2/ssl/SSL_set_tlsext_status_type.html
+     *
+     * @see #TLSEXT_STATUSTYPE_host_name
+     * @see #TLSEXT_STATUSTYPE_ocsp
+     */
+    public static native boolean setStatusType(long ssl, int type);
```

Comment:
If we don't care about the exact details of `SSL_set_tlsext_status_type()` we can make it `setOcspEnabled(long ssl, boolean enabled)`